presto-common/src/main/java/com/facebook/presto/common/type/LongEnumType.java

```diff
@@ -100,9 +100,10 @@ public LongEnumMap(@JsonProperty("enumMap") Map<String, Long> enumMap)
 
         public Optional<String> getKeyForValue(Long value)
         {
-            if (!isFlippedEnumComputed.getAndSet(true)) {
+            if (!isFlippedEnumComputed.get()) {
                 this.flippedEnumMap = this.enumMap.entrySet().stream()
```

Comment:
now that I'm reading and thinking more about this, this lazy initialization in a multi-threaded environment is a non-trivial problem 
* https://stackoverflow.com/questions/8297705/how-to-implement-thread-safe-lazy-initialization
* https://android.jlelse.eu/lazy-initialisation-whats-a-correct-implementation-64c4638561e

especially since inside presto-commons we don't have access to Guava or Apache Commons which give us utility methods to solve this, I'm tempted to remove the lazy-initialization and keep it in the constructor to avoid the complexity here. @rongrong what do you think?

Reply:
I am not doing enough Java to know if this is the right approach: but is that the kind of situation where we can use the `synchronized` keyword? 

Something like:
```
if (!isFlippedEnumComputed.get()) {
    synchronized(..?) {
        if (!isFlippedEnumComputed.get()) { // second check in case it was enabled between the first check and the synchronized stmt
        }
    }
}
```

Reply:
Yes I considered that one. But reading through the examples in the second article which looked correct to me and turned out to be unsafe, I realized I don't have good intuition for how to do concurrency safely in Java, and since we didn't have IMO a strong need to do this lazy initialization in the first place, I'd rather sidestep that complexity altogether.

Reply:
Oh I did not check the links, I see it mentions the synchronised thing. But I totally agree with you, synchronization can be so tricky... so if you can avoid it...